src/app/dashboard/list/list.component.ts

```diff
@@ -0,0 +1,113 @@
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import { Component, Input, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { FormGroupDirective } from '@angular/forms';
+import { ICategory } from 'src/app/core/interfaces/category';
+import { ITask } from 'src/app/core/interfaces/task';
+import { Key } from 'src/app/core/constants/keyboard-keys';
+import { TaskService } from 'src/app/core/services/task-service/task-service';
+import { ViewChild } from '@angular/core';
+
+@Component({
+  selector: 'app-list',
+  templateUrl: './list.component.html',
+  styleUrls: ['./list.component.css']
+})
+export class ListComponent implements OnInit {
+  @Input() category!: ICategory;
+
+  @ViewChild(FormGroupDirective, { static: false })
+  formDirective!: FormGroupDirective;
+
+  taskName = new FormControl('', [Validators.required]);
```

Comment:
Please use FormBuilder when creating forms

https://angular.io/guide/reactive-forms#using-the-formbuilder-service-to-generate-controls